crates/core/tedge_agent/src/agent.rs

```diff
@@ -182,7 +182,13 @@ impl Agent {
         let signal_actor_builder = SignalActor::builder(&runtime.get_handle());
 
         // Health actor
-        let health_actor = HealthMonitorBuilder::new(TEDGE_AGENT, &mut mqtt_actor_builder);
+        let service_topic_id = self.config.mqtt_device_topic_id.to_service_topic_id("tedge-agent")
+            .with_context(|| format!("Device topic id {} currently needs default scheme, e.g: 'device/DEVICE_NAME//'", self.config.mqtt_device_topic_id))?;
```

Comment:
This will fail if `mqtt_device_topic_id` doesn't fit default scheme. Is there a better way to handle this?

Reply:
The agent should be given an entity topic identifier for itself (as an argument). `mqtt_device_topic_id.to_service_topic_id("tedge-agent")` should be used only as a default value.

Reply:
I think this is okay-ish for this PR.